src/hooks/useApi.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState, useCallback } from 'react';
+
+type ApiResponse<T, P extends object> = {
+  data: T | null;
+  isLoading: boolean;
+  error: string | null;
+  makeRequest: (params: P) => Promise<void>;
+};
+
+export function useApi<T, P extends object>(
+  fetchFunction: (params: P) => Promise<T>,
+  params: P
+): ApiResponse<T, P> {
+  const [data, setData] = useState<T | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const wrappedFunction = useCallback(
+    async (params: P) => {
+      setIsLoading(true);
+      setError(null);
+
+      try {
+        const result = await fetchFunction(params);
+        setData(result);
+      } catch (err) {
+        console.error(err);
+        setError('데이터 로딩 실패');
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [fetchFunction]
+  );
+
+  useEffect(() => {
+    wrappedFunction(params);
+  }, [JSON.stringify(params), wrappedFunction]);
```

Comment:
P3:

왜 이렇게 하셨을까요? 가능하면 param을 deps array에 넣어주시고
이를 통해 구현하고자 하셨던 바는 다른방식으로 구현하시는것이 더 좋을 것 같아요

```suggestion
  }, [params, wrappedFunction]);
```